client/my-sites/earn/memberships/products.jsx

```diff
@@ -352,6 +193,8 @@ class MembershipsProductsSection extends Component {
 						{ this.renderEllipsisMenu( product.ID ) }
 					</CompactCard>
 				) ) }
+				{ this.renderAddNewPlanDialog() }
+				{ this.renderEditPlanDialog() }
```

Comment:
Does it make sense to have both of these? It appears as those they are identical functions/methods except one uses `product={this.state.product}` while one uses (well _should use_) `product={null}`. Both of which can be inferred from `this.state.showDialog` being `editPlan` or `addNewPlan`.

Reply:
You're right; but that was giving me the same flash of Edit modal to Add New modal during closing that is the original problem that prompted this PR. But maybe I'm not doing something in the right order – I'll take another look.

Reply:
I simplified the behavior in 691c3c6 so only one dialog is rendered at a time.